ksql-execution/src/main/java/io/confluent/ksql/execution/util/ExpressionTypeManager.java

```diff
@@ -374,27 +390,20 @@ public Void visitFunctionCall(
         return null;
       }
 
-      if (node.getName().name().equalsIgnoreCase(FetchFieldFromStruct.FUNCTION_NAME)) {
-        process(node.getArguments().get(0), expressionTypeContext);
-        final Schema firstArgSchema = expressionTypeContext.getSchema();
-        final String fieldName = ((StringLiteral) node.getArguments().get(1)).getValue();
-        if (firstArgSchema.field(fieldName) == null) {
-          throw new KsqlException(String.format("Could not find field %s in %s.",
-              fieldName,
-              node.getArguments().get(0).toString()));
-        }
-        final Schema returnSchema = firstArgSchema.field(fieldName).schema();
-        expressionTypeContext.setSchema(returnSchema);
-      } else {
-        final UdfFactory udfFactory = functionRegistry.getUdfFactory(node.getName().name());
-        final List<Schema> argTypes = new ArrayList<>();
-        for (final Expression expression : node.getArguments()) {
-          process(expression, expressionTypeContext);
-          argTypes.add(expressionTypeContext.getSchema());
-        }
-        final Schema returnSchema = udfFactory.getFunction(argTypes).getReturnType(argTypes);
-        expressionTypeContext.setSchema(returnSchema);
+      final UdfFactory udfFactory = functionRegistry.getUdfFactory(node.getName().name());
+      if (udfFactory.isInternal()) {
```

Comment:
this smells a little bit to me - if I were reading this code without the context of this PR I would ask "why do we bother having 'internal functions' if we can't use them anywhere?". The fact that we implement the struct fetcher as a UDF shouldn't leak into the FunctionCall handling code.

This might be an unfortunate and unavoidable side-effect of what you pointed out in the description that we still implement it as a UDF, so at a minimum we should add an inline comment explaining what an "internalFunction" is and that it is, in reality, not a valid function call at all.

Reply:
AIUI, the only internal udf we have is FetchFieldFromStruct, so once we get rid of that, we can also get rid of this concept of "internal udf" and get rid of this check.

Reply:
Yeah, I call out that we may want to move away from a UDF for structs in the description. I've raised #3676 to track.  It's one for the backlog.

AS for this... I'll add a comment for now.  This is still a step in the right direction.